common/modal.js

```diff
@@ -181,22 +181,35 @@
             getDisabledTuples:  params.getDisabledTuples
         };
 
-        // since this is currently used for single select mode, the isSelected will always be true
-        function ok(tuples, isSelected) {
-            if (params.selectMode != modalBox.multiSelectMode) $uibModalInstance.close(tuples[0]);
+        function onRowClick(tuples, isSelected) {
+            switch (params.selectMode) {
+                case modalBox.multiSelectMode:
+                    if (params.onRowClick) {
+                        // if it returns false, then we should disable submit button
+                        vm.disableSubmit = (params.onRowClick(getMultiSelectionResult()) === false);
+                    }
+                    break;
+                default:
+                    // for single select isSelected is always true
+                    $uibModalInstance.close(tuples[0]);
+            }
         }
 
-        /**
-         * Will call the close modal with the appropriate results.
-         * If we had the matchNotNull, then we just need to pass that attribute.
-         */
-        function submitMultiSelection() {
+        function getMultiSelectionResult() {
             var res = vm.tableModel.selectedRows;
             if (!Array.isArray(res)) res = [];
             if (vm.tableModel.matchNotNull) {
                 res = {matchNotNull: true};
             }
-            $uibModalInstance.close(res);
+            return res;
+        }
+
+        /**
+         * Will call the close modal with the appropriate results.
+         * If we had the matchNotNull, then we just need to pass that attribute.
+         */
+        function submitMultiSelection() {
+            $uibModalInstance.close(getMultiSelectionResult());
```

Comment:
How is the return value of `getMultiSelectionResult()` used int he modal close callback (which is `modalDataChanged(scope, true)`, on line 1092 of `faceting.js`)?

Reply:
If you look at the recordsetSelectFaceting, it has an extra checkbox for matching not-null values. So apart from `selectedRows` the value of `matchNotNull` is important too. The `getMultiSelectionResult` is taking care of that. And then the `modalDataChanged` is looking for those two variables.